File: asml/LC80RemoveDuplicatesFromSortedArrayII.py
class Solution(object):
    def removeDuplicates(self, nums):
        if not nums or len(nums) == 1 or len(nums) == 2:
            return
        start1 = 0
        start2 = 1
        for i in range(2, len(nums)):
            if nums[i] == nums[start1] == nums[start2]:
                pass
            elif nums[i] == nums[start2] and nums[i] != nums[start1]:
                start1 += 1
                start2 += 1
                nums[start2] = nums[i]
            elif nums[i] != nums[start2]:
                start1 += 1
                start2 += 1
                nums[start2] = nums[i]
        return start2+1

File: asml/LC80RemoveDuplicatesFromSortedArrayII.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        if (nums.size() <= 2) return nums.size();
        int rear = 1;
        for(int i = 2; i < nums.size(); i++) {
            if (nums[i] != nums[rear-1]) {
                nums[++rear] = nums[i];
            }
        }
        return rear+1;
    }
};
